Dockerfile

```diff
@@ -17,6 +17,9 @@ RUN apt-get update \
         unzip \
         vim \
         wget \
+        build-essential \
+        gcc-x86-64-linux-gnu \
+        python3-dev \
```

Comment:
@hectorpal These needed anymore?

Reply:
I tried to explain that without them, MACQ installation fails.

I mean, have you tried installing MACQ? Anything compiling things might fail.
I was wondering if it always do that. It's not unlikely they were missing precompiled wheels and triggered a compilation. 

No clue

Reply:
Hrmz...news to me. MACQ's meant to be just a straight-forward python lib. Maybe it's one of the requirements that aren't getting fully done (tarski, kissat, etc.)

**Update**

Ya, it's python-sat that chokes:

![image](https://user-images.githubusercontent.com/225019/174306731-7f381889-153a-4fa7-8f81-ce27709e3c3c.png)

Reply:
https://pysathq.github.io/installation/

Reply:
@FlorianPommerening / @hectorpal : What are your thoughts on having this in the Dockerfile -vs- install script for macq?

Reply:
Making this part of the install script is a bit tricky. What do you do in the uninstall script then? If you apt-remove it, that could be an issue if it was already installed before installing planutils. I think g++, build-essentials and Python 3 are all "generic" enough to install them together with plan-utils.

This is not a general solution, though. I wouldn't know how to deal with a more specific apt package that some other planutils package needs as a dependency.

Reply:
Ya, the thing is that if users install without docker, then they'll be on their own to make sure those packages are there. Maybe there's an in-between where...

1. We include it in the Docker
2. We test in the install script for the pre-req's (**any ideas on how to do this nicely in bash?**)
3. We abort the installation and point the user to installing it themselves.

I think this may be similar to getting singularity installed (cf. #79 ). I.e., explain thing to the user and leave it up to them to install / maintain when it might be removed.

Reply:
A bit of convoluted solution is to create "wheels" for that version of Ubuntu. Then pip install from those wheels.

We did it for an internal project. I could look it up.

We can recommend that was a requirement for packagers: "You should be able to install without compiling. Singularity is covered. Here is a Docker image if you want to generate binaries". I did that for generating the FF binary, changing docker to pip install planutils from current dir: `pip -e .`

Reply:
Ok, will punt for now -- https://github.com/AI-Planning/macq/issues/186 -- and keep the changes to the docker setup.